### ISSUE TITLE
Strange Grad Curve

### ISSUE THREAD

**Issue by lavinal712** [bluorion-com/ZClip#4]
![Image](https://github.com/user-attachments/assets/8d0e09d7-cd9e-437c-aa56-86d5714b2bfc)

Could you explain why there's such a strange curve? In my opinion, it looks like the model has collapsed.

**Comment by akanyaani** [comment]
@lavinal712 This strange curve appears because, to test ZClip's robustness, we pre-trained a 1B LLaMA model using an unusually high learning rate range of 1e-3 to 5e-3. As you can see, the model with just constant clipping diverges rapidly, whereas the one using ZClip remains stable and continues training effectively.
